Mathlib/Topology/UniformSpace/Basic.lean

```diff
@@ -0,0 +1,1966 @@
+    {b : Set (β × β)} (ha : a ∈ 𝓤 α) (hb : b ∈ 𝓤 β) :
+    { p : (α × β) × α × β | (p.1.1, p.2.1) ∈ a ∧ (p.1.2, p.2.2) ∈ b } ∈ 𝓤 (α × β) := by
+  rw [uniformity_prod]; exact inter_mem_inf (preimage_mem_comap ha) (preimage_mem_comap hb)
+#align mem_uniform_prod mem_uniform_prod
+
+theorem tendsto_prod_uniformity_fst [UniformSpace α] [UniformSpace β] :
+    Tendsto (fun p : (α × β) × α × β => (p.1.1, p.2.1)) (𝓤 (α × β)) (𝓤 α) :=
+  le_trans (map_mono inf_le_left) map_comap_le
+#align tendsto_prod_uniformity_fst tendsto_prod_uniformity_fst
+
+theorem tendsto_prod_uniformity_snd [UniformSpace α] [UniformSpace β] :
+    Tendsto (fun p : (α × β) × α × β => (p.1.2, p.2.2)) (𝓤 (α × β)) (𝓤 β) :=
+  le_trans (map_mono inf_le_right) map_comap_le
+#align tendsto_prod_uniformity_snd tendsto_prod_uniformity_snd
+
+theorem uniformContinuous_fst [UniformSpace α] [UniformSpace β] :
+    UniformContinuous fun p : α × β => p.1 :=
+  tendsto_prod_uniformity_fst
+#align uniform_continuous_fst uniformContinuous_fst
+
+theorem uniformContinuous_snd [UniformSpace α] [UniformSpace β] :
+    UniformContinuous fun p : α × β => p.2 :=
+  tendsto_prod_uniformity_snd
+#align uniform_continuous_snd uniformContinuous_snd
+
+variable [UniformSpace α] [UniformSpace β] [UniformSpace γ]
+
+theorem UniformContinuous.prod_mk {f₁ : α → β} {f₂ : α → γ} (h₁ : UniformContinuous f₁)
+    (h₂ : UniformContinuous f₂) : UniformContinuous fun a => (f₁ a, f₂ a) := by
+  rw [UniformContinuous, uniformity_prod]
+  exact tendsto_inf.2 ⟨tendsto_comap_iff.2 h₁, tendsto_comap_iff.2 h₂⟩
+#align uniform_continuous.prod_mk UniformContinuous.prod_mk
+
+theorem UniformContinuous.prod_mk_left {f : α × β → γ} (h : UniformContinuous f) (b) :
+    UniformContinuous fun a => f (a, b) :=
+  h.comp (uniformContinuous_id.prod_mk uniformContinuous_const)
+#align uniform_continuous.prod_mk_left UniformContinuous.prod_mk_left
+
+theorem UniformContinuous.prod_mk_right {f : α × β → γ} (h : UniformContinuous f) (a) :
+    UniformContinuous fun b => f (a, b) :=
+  h.comp (uniformContinuous_const.prod_mk uniformContinuous_id)
+#align uniform_continuous.prod_mk_right UniformContinuous.prod_mk_right
+
+theorem UniformContinuous.prod_map [UniformSpace δ] {f : α → γ} {g : β → δ}
+    (hf : UniformContinuous f) (hg : UniformContinuous g) : UniformContinuous (Prod.map f g) :=
+  (hf.comp uniformContinuous_fst).prod_mk (hg.comp uniformContinuous_snd)
+#align uniform_continuous.prod_map UniformContinuous.prod_map
+
+theorem toTopologicalSpace_prod {α} {β} [u : UniformSpace α] [v : UniformSpace β] :
+    @UniformSpace.toTopologicalSpace (α × β) instUniformSpaceProd =
+      @instTopologicalSpaceProd α β u.toTopologicalSpace v.toTopologicalSpace :=
+  rfl
+#align to_topological_space_prod toTopologicalSpace_prod
+
+/-- A version of `UniformContinuous.inf_dom_left` for binary functions -/
+theorem uniformContinuous_inf_dom_left₂ {α β γ} {f : α → β → γ} {ua1 ua2 : UniformSpace α}
```

Comment:
Should this also be renamed for dot notation? (Same question for the next lemma)

Reply:
Probably a good idea, but let's do it in a follow-up PR.